src/main/java/seedu/address/logic/commands/cart/CartCommand.java

```diff
@@ -0,0 +1,80 @@
+package seedu.address.logic.commands.cart;
+
+import java.util.Optional;
+
+import seedu.address.commons.util.CollectionUtil;
+import seedu.address.logic.commands.Command;
+import seedu.address.model.Cart;
+import seedu.address.model.ReadOnlyIngredientList;
+import seedu.address.model.ingredient.UniqueIngredientList;
+
+/**
+ * Represents an Cart command with hidden internal logic and the ability to be executed.
+ */
+public abstract class CartCommand extends Command {
+
+    public static final String COMMAND_CATEGORY = "cart";
+
+    /**
+     * Stores the details to edit the cart with.
+     */
+    public static class EditCartDescriptor {
```

Comment:
Actually it's supposed to be EditIngredientDescriptor, to edit the ingredients inside the cart. Same for inventory. But I don't think we really need this, because we can directly get the cart to add/remove ingredients, we don't have to edit ingredients.

Reply:
Oh, cos from what i gathered, edit just means add or delete, not editing the values of individual ingredients

Reply:
The way the original address book did it, they had an `EditPersonDescriptor` for editing people in the list. So the parallel idea is to have `EditRecipeDescriptor` for editing recipes in the cookbook, and `EditIngredientDescriptor` for editing ingredients in the inventory/cart.

Reply:
Ok! will resolve in the next PR